src/components/LoanCards/KvClassicLoanCardContainer.vue

```diff
@@ -274,7 +285,68 @@ export default {
 		},
 		setCookie(name, value, options) {
 			return this.cookieStore.set(name, value, options);
-		}
+		},
+		async fetchLoanActivity() {
+			const activities = [];
+
+			const activityResponse = await this.apollo.query({
+				query: loanActivitiesQuery,
+				variables: { loanId: this.loanId }
+			});
+
+			const lendingActionsData = activityResponse.data?.lend?.loan?.lendingActions?.values ?? [];
+			const commentsData = activityResponse.data?.lend?.loan?.comments?.values ?? [];
+
+			lendingActionsData.forEach(action => {
+				const actionDate = new Date(action?.latestSharePurchaseDate).toDateString();
+				if (!activities.some(activity => activity.key === actionDate)) {
+					activities.push({
+						key: actionDate,
+						data: [],
+					});
+				}
+				if (action?.lender.name) {
+					const dataObject = activities.find(activity => activity.key === actionDate);
+					dataObject?.data.push({
+						lenderName: action.lender.name,
+						lenderImage: action.lender.image?.url,
+						text: `${action.lender.name} lent $${parseFloat(action.shareAmount).toFixed()}`,
+						date: action.latestSharePurchaseDate,
+						type: action.__typename, // eslint-disable-line no-underscore-dangle
+					});
+				}
+			});
+
+			commentsData.forEach(comment => {
+				const commentDate = new Date(comment?.date).toDateString();
+				if (!activities.some(activity => activity.key === commentDate)) {
+					activities.push({
+						key: commentDate,
+						data: [],
+					});
+				}
+				if (comment?.authorName) {
+					const dataObject = activities.find(activity => activity.key === commentDate);
+					dataObject?.data.push({
+						lenderName: comment.authorName,
+						lenderImage: comment.authorLendingAction?.lender.image?.url,
```

Comment:
Looks like this needs one more chaining: `comment.authorLendingAction?.lender?.image?.url`